src/main/java/de/tum/in/www1/artemis/repository/TextBlockRepository.java

```diff
@@ -30,4 +32,45 @@
     Set<TextBlock> findAllBySubmissionIdIn(Set<Long> submissionIdList);
 
     void deleteAllBySubmission_Id(Long submissionId);
+
+    /**
+     * Interface to define return type for countOtherBlocksInClusterForSubmission
+     */
+    interface TextBlockCount {
+
+        String getBlockId();
+
+        Long getNumberOfOtherBlocks();
+    }
+
+    /**
+     * For the given Submission `id` returns a list of raw object array representing two columns.
+     * First index/column corresponds to the TextBlock `id` while the second one corresponds to
+     * the number of other blocks in the same cluster as given block with id = `id`.
+     * For all TextBlock's of the Submission with the given `id`
+     * finds their respective cluster and retrieves the number of other blocks in the same cluster
+     * @param id the id of the Submission
+     * @return the number of other TextBlock's in the same cluster as the block with given `id`
+     */
+    @Query("""
+            SELECT tb.id as blockId, COUNT(DISTINCT tball.id) as numberOfOtherBlocks
+            FROM TextSubmission s
+            LEFT JOIN TextBlock tb ON s.id = tb.submission.id
+            LEFT JOIN TextCluster tc ON tb.cluster.id = tc.id
+            LEFT JOIN TextBlock tball ON tc.id = tball.cluster.id AND tball.id <> tb.id
+            WHERE s.id = :#{#id}
+            GROUP BY tb.id
+            """)
+    List<TextBlockCount> countOtherBlocksInClusterForSubmission(@Param("id") Long id);
```

Comment:
`id`-->`submissionId`

Reply:
Fixed in [commit](https://github.com/ls1intum/Artemis/pull/3456/commits/c850023ccd3f5d0b2f1f57937beaebaada6d5300)